src/payment/strategies/digitalriver/digitalriver-payment-strategy.spec.ts

```diff
@@ -0,0 +1,259 @@
+        let options: PaymentInitializeOptions;
+        let onErrorCallback: (error: OnCancelOrErrorResponse) => {};
+
+        beforeEach(() => {
+            options = getInitializeOptions();
+            jest.spyOn(store.getState().billingAddress, 'getBillingAddressOrThrow').mockReturnValue(getBillingAddress());
+            jest.spyOn(store.getState().customer, 'getCustomer').mockReturnValue(customer);
+            jest.spyOn(digitalRiverScriptLoader, 'load').mockReturnValue(Promise.resolve(digitalRiverLoadResponse));
+            jest.spyOn(digitalRiverLoadResponse, 'createDropin').mockReturnValue(digitalRiverComponent);
+        });
+
+        it('returns the state', async () => {
+            await expect(strategy.initialize(options)).resolves.toBe(store.getState());
+        });
+
+        it('loads DigitalRiver script', async () => {
+            await expect(strategy.initialize(options)).resolves.toBe(store.getState());
+            expect(digitalRiverScriptLoader.load).toHaveBeenCalled();
+        });
+
+        it('calls onSuccess callback from DigitalRiver', async () => {
+            jest.spyOn(digitalRiverLoadResponse, 'createDropin').mockImplementation(({onSuccess}) => {
+                onSuccess({
+                    source: {
+                        id: '1',
+                        reusable: false,
+                    },
+                    readyForStorage: true,
+                });
+
+                return digitalRiverComponent;
+            });
+            await strategy.initialize(options);
+            expect(true).toBeTruthy();
+        });
+
+        it('calls onReady callback from DigitalRiver', async () => {
+            jest.spyOn(digitalRiverLoadResponse, 'createDropin').mockImplementation(({onReady}) => {
+                onReady({
+                    paymentMethodTypes: ['creditCard', 'paypal'],
+                });
+
+                return digitalRiverComponent;
+            });
+            await strategy.initialize(options);
+            expect(true).toBeTruthy();
+        });
+
+        it('calls onError callback from DigitalRiver', async () => {
+            jest.spyOn(digitalRiverLoadResponse, 'createDropin').mockImplementation(({onError}) => {
+                onErrorCallback = onError;
+
+                return digitalRiverComponent;
+            });
+            await strategy.initialize(options);
+            onErrorCallback({
+                errors: [{
+                    code: 'code',
+                    message: 'message',
+                }],
+            });
+            expect(options.digitalriver?.onError).toBeCalled();
+            expect(digitalRiverLoadResponse.createDropin).toBeCalled();
+        });
+
+        it('throws an error when load response is empty or not provided', () => {
+            const promise = strategy.initialize(options);
+            jest.spyOn(digitalRiverScriptLoader, 'load').mockReturnValue(Promise.resolve(undefined));
```

Comment:
Move the spy's mocks at the beginning of the test. Apply this for all this file.